### PR TITLE
Issue #318 - add support for processing JSON with unrecognized elements

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: fhir-model/src/main/java/com/ibm/fhir/model/parser/FHIRAbstractParser.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package com.ibm.fhir.model.parser;
```
Comment (prb112) [inline fhir-model/src/main/java/com/ibm/fhir/model/parser/FHIRAbstractParser.java]: Copyright header

Comment (JohnTimm) [reply fhir-model/src/main/java/com/ibm/fhir/model/parser/FHIRAbstractParser.java]: Added.

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchTest.java [hunk 2]
```diff
@@ -911,7 +917,7 @@ public void test_SearchObservationFilteredSearchParameter1_preferBogus()
             throws Exception {
         // 'category' search parameter is filtered out for tenant1.
```
Comment (prb112) [inline fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchTest.java]: should this comment be updated?

File: fhir-config/src/main/java/com/ibm/fhir/config/FHIRConfiguration.java [hunk 3]
```diff
@@ -28,7 +28,8 @@
     public static final String PROPERTY_DATASTORE_ID_HEADER_NAME = "fhirServer/core/datastoreIdHeaderName";
     public static final String PROPERTY_DEFAULT_TENANT_ID = "fhirServer/core/defaultTenantId";
     public static final String PROPERTY_DEFAULT_PRETTY_PRINT = "fhirServer/core/defaultPrettyPrint";
-    public static final String PROPERTY_HANDLING = "fhirServer/core/handling";
+    public static final String PROPERTY_DEFAULT_HANDLING = "fhirServer/core/defaultHandling";
+    public static final String PROPERTY_ALLOW_CLIENT_HANDLING_PREF = "fhirServer/core/allowClientHandlingPref";
```
Comment (prb112) [inline fhir-config/src/main/java/com/ibm/fhir/config/FHIRConfiguration.java]: update to documentation in this PR? or in a subsequent PR?

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchTest.java (hunk 2 above)
Comment (JohnTimm) [reply fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchTest.java]: removed comment

File: fhir-config/src/main/java/com/ibm/fhir/config/FHIRConfiguration.java (hunk 3 above)
Comment (JohnTimm) [reply fhir-config/src/main/java/com/ibm/fhir/config/FHIRConfiguration.java]: subsequent

Comment (lmsurpre) [reply fhir-config/src/main/java/com/ibm/fhir/config/FHIRConfiguration.java]: I'd vote to get it done in this PR

File: fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java [hunk 4]
```diff
@@ -2287,7 +2290,8 @@ private void generateJsonParser(String basePath) {
         
         cb.annotation("NotThreadSafe");
         cb.annotation("Generated", quote("com.ibm.fhir.tools.CodeGenerator"));
-        cb._class(mods("public"), "FHIRJsonParser", null, implementsInterfaces("FHIRParser"));
+//      cb._class(mods("public"), "FHIRJsonParser", null, implementsInterfaces("FHIRParser"));
```
Comment (lmsurpre) [inline fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java]: remove this line?

File: fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java [hunk 5]
```diff
@@ -1888,7 +1889,8 @@ private void generateXMLParser(String basePath) {
         
         cb.annotation("NotThreadSafe");
         cb.annotation("Generated", quote("com.ibm.fhir.tools.CodeGenerator"));
-        cb._class(mods("public"), "FHIRXMLParser", null, implementsInterfaces("FHIRParser"));
+//      cb._class(mods("public"), "FHIRXMLParser", null, implementsInterfaces("FHIRParser"));
```
Comment (lmsurpre) [inline fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java]: remove this line?

File: fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java [hunk 6]
```diff
@@ -1872,7 +1872,8 @@ private void generateXMLParser(String basePath) {
         cb._import("javax.xml.stream.XMLStreamReader");
         cb.newLine();
         
-        cb._import("com.ibm.fhir.model.parser.FHIRParser");
+//      cb._import("com.ibm.fhir.model.parser.FHIRParser");
```
Comment (lmsurpre) [inline fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java]: remove this line?

File: fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java (hunk 4 above)
Comment (JohnTimm) [reply fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java]: removed

File: fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java (hunk 5 above)
Comment (JohnTimm) [reply fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java]: removed

File: fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java (hunk 6 above)
Comment (JohnTimm) [reply fhir-tools/src/main/java/com/ibm/fhir/tools/CodeGenerator.java]: removed

**Pull request merged**